comit/src/expiries/config.rs

```diff
@@ -0,0 +1,215 @@
+    beta_block_time: u16,
+    alpha_mine_fund_within_n_blocks: u8,
+    beta_mine_fund_within_n_blocks: u8,
+    alpha_mine_redeem_within_n_blocks: u8,
+    beta_mine_redeem_within_n_blocks: u8,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ActorType {
+    Bot,
+    Human,
+}
+
+impl ActorType {
+    fn period_to_act(&self) -> Duration {
+        match self {
+            ActorType::Bot => 15.minutes(),
+            ActorType::Human => 60.minutes(),
+        }
+    }
+}
+
+impl Config {
+    /// The duration of time it takes for Alice to act.
+    pub fn alice_to_act(&self) -> Duration {
+        self.alice.period_to_act()
+    }
+
+    /// The duration of time it takes for Bob to act.
+    pub fn bob_to_act(&self) -> Duration {
+        self.bob.period_to_act()
+    }
+
+    /// The duration of time it takes for the alpha fund transaction to be
+    /// mined into the blockchain.
+    pub fn mine_alpha_fund_transaction(&self) -> Duration {
+        let n = self.alpha_mine_fund_within_n_blocks;
+        let block_time = self.alpha_block_time;
+
+        time_to_mine_n_blocks(n, block_time)
+    }
+
+    /// The duration of time it takes for the beta fund transaction to be
+    /// mined into the blockchain.
+    pub fn mine_beta_fund_transaction(&self) -> Duration {
+        let n = self.beta_mine_fund_within_n_blocks;
+        let block_time = self.beta_block_time;
+
+        time_to_mine_n_blocks(n, block_time)
+    }
+
+    /// The duration of time it takes for the alpha redeem transaction to be
+    /// mined into the blockchain.
+    pub fn mine_alpha_redeem_transaction(&self) -> Duration {
+        let n = self.alpha_mine_redeem_within_n_blocks;
+        let block_time = self.alpha_block_time;
+
+        time_to_mine_n_blocks(n, block_time)
+    }
+
+    /// The duration of time it takes for the beta redeem transaction to be
+    /// mined into the blockchain.
+    pub fn mine_beta_redeem_transaction(&self) -> Duration {
+        let n = self.beta_mine_redeem_within_n_blocks;
+        let block_time = self.beta_block_time;
+
+        time_to_mine_n_blocks(n, block_time)
+    }
+
+    /// The duration of time it takes for a transaction to reach finality on the
+    /// alpha ledger.
+    pub fn finality_alpha(&self) -> Duration {
+        let n = self.alpha_required_confirmations;
+        let block_time = self.alpha_block_time;
+
+        time_to_mine_n_blocks(n, block_time)
+    }
+
+    /// The duration of time it takes for a transaction to reach finality on the
+    /// beta ledger.
+    pub fn finality_beta(&self) -> Duration {
+        let n = self.beta_required_confirmations;
+        let block_time = self.beta_block_time;
+
+        time_to_mine_n_blocks(n, block_time)
+    }
+}
+
+fn bitcoin_required_confirmations() -> u8 {
+    BITCOIN_CONFIRMATIONS
+}
+
+fn ethereum_required_confirmations() -> u8 {
+    ETHEREUM_CONFIRMATIONS
+}
+
+// Time to mine n blocks is governed by a Poisson distribution. As an
```

Comment:
This is outdated now, no?

Reply:
No its not outdated, its still governed by this we just do not calculate it. I include the note in case we wish to calculate this at some later date.